src/components/SearchList.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Resource } from '../store/resourcesSlice';
+import PreviewCard from './PreviewCard';
+import { RootState } from '../store/reducers';
+
+function fuzzysearch(needle, haystack) {
+  const hlen = haystack.length;
+  const nlen = needle.length;
+  let score = 0;
+  if (nlen > hlen) {
+    return score;
+  }
+  if (nlen === hlen) {
+    return Number.MAX_SAFE_INTEGER;
+  }
+  let i = 0;
+  let j = 0;
+  outer: for (i = 0, j = 0; i < nlen; i++) {
+    const nch = needle.charCodeAt(i);
+    while (j < hlen) {
+      if (haystack.charCodeAt(j++) === nch) {
+        score++;
+        continue outer;
+      }
+    }
+    return score;
+  }
+  return score;
+}
+
+function SearchList(props) {
+  const resources = useSelector((state: RootState) => state.resources);
```

Comment:
Can rewrite these next 3 lines as:
```
const resources = useSelector((state: RootState) => {
  return Object.keys(state.resources).map<Resource>((r: string) => state.resources[r]);
});
```

(`.map` by default returns a new array, so I think it's better practice to create the array that way rather than using map to manually append to a different array)